packages/ilib-lint/test/ResourceSentenceEnding.test.js

```diff
@@ -0,0 +1,1521 @@
+            file: "a/b/c.xliff"
+        });
+        // Should trigger because Japanese target has English period instead of maru
+        expect(actual).toBeTruthy();
+        expect(actual.description).toContain("Sentence ending punctuation should be \"。\" for ja-JP locale");
+    });
+
+    test("Japanese question mark is converted to fullwidth", () => {
+        expect.assertions(2);
+
+        const rule = new ResourceSentenceEnding();
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "question.test",
+            sourceLocale: "en-US",
+            source: "What is this?",
+            targetLocale: "ja-JP",
+            target: "これは何ですか？",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        // Should not trigger because Japanese target already has correct question mark (？)
+        expect(actual === null).toBeTruthy();
+    });
+
+    test("Japanese question mark triggers warning if not fullwidth", () => {
+        expect.assertions(3);
+
+        const rule = new ResourceSentenceEnding();
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "question.test",
+            sourceLocale: "en-US",
+            source: "What is this?",
+            targetLocale: "ja-JP",
+            target: "これは何ですか?",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        // Should trigger because Japanese target has English question mark instead of Japanese one
+        expect(actual).toBeTruthy();
+        expect(actual.description).toContain("Sentence ending punctuation should be \"？\" for ja-JP locale");
+    });
+
+    test("Japanese exclamation mark is converted to fullwidth", () => {
+        expect.assertions(2);
+
+        const rule = new ResourceSentenceEnding();
+        expect(rule).toBeTruthy();
+
+        const resource = new ResourceString({
+            key: "exclamation.test",
+            sourceLocale: "en-US",
+            source: "This is amazing!",
+            targetLocale: "ja-JP",
+            target: "これは素晴らしいです！",
+            pathName: "a/b/c.xliff"
+        });
+        const actual = rule.matchString({
+            source: resource.getSource(),
+            target: resource.getTarget(),
+            resource,
+            file: "a/b/c.xliff"
+        });
+        // Should not trigger because Japanese target already has correct exclamation mark (！)
+        expect(actual === null).toBeTruthy();
+    });
+
+    test("Japanese exclamation mark triggers warning if not fullwidth", () => {
+        expect.assertions(3);
+
```

Comment:
Question: 
For example, in Japanese, if I want to use the default full-width period '。' but avoid using full-width for other punctuation marks like question marks or exclamation points, I need to redefine the entire set (including the period) in the configuration file. am I correct?

Reply:
I modified it so that the default punctuation is first and anything specified in the config file overrides it. That way, you only have to specify the punctuation that is different than the standard default (English) ones.